query-graphs-js/src/querygraph.ts

```diff
@@ -965,6 +965,11 @@ class GraphBuilderFromSchema extends GraphBuilder {
     return !!metadata && field.hasAppliedDirective(directiveFct(metadata));
   }
 
+  private isExternal(field: FieldDefinition<any>): boolean {
+    const metadata = federationMetadata(this.schema);
+    return !!metadata && metadata.isFieldExternal(field);
```

Comment:
@pcmanus 
I followed the pattern you had above, but is there a reason we shouldn't just assert `metadata` here?

Reply:
Yes. We build `QueryGraph` using this code both for subgraph schema _and_ for the supergraph API schema. The former will always have a `metadata` but the later won't, but we don't want to hard fail on it.

If we were to guard the calling of those methods with `this.isFederatedSubgraph`, then yes, we could assert `metadata` at that point, but not sure it would be much cleaner.